### PR TITLE
EPMDJ-6670: Update docs page layout, sidebar, ToC styles

### PR DESCRIPTION
**Pull request by IliaSolovev**
Issue: EPMDJ-6670

### CODE REVIEW

File: src/theme/DocItem/index.tsx [hunk 1]
```diff
@@ -78,99 +58,26 @@ function DocItem(props: Props): JSX.Element {
...
-                </div>
-              </div>
-            )}
-            <div className="margin-vert--lg">
-              <DocPaginator metadata={metadata} />
-            </div>
+      <main className="col-span-7">
+        <article>
+          {!hideTitle && (
+            <header>
```
Comment (RomanAchilov) [inline src/theme/DocItem/index.tsx]: I'm not sure what to wrap in the header here

example
```
<header />
<main>
  <article>
    <h1 />
   </article>
</main>
<footer />

File: src/theme/DocItem/styles.scss [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+  margin-top: 24px;
+
+  li {
+    padding-left: 8px;
+    margin: 0;
+
+    .table-of-contents__link {
+      display: inline-block;
+      margin-bottom: 8px;
+      font-size: 13px;
```
Comment (RomanAchilov) [inline src/theme/DocItem/styles.scss:13]: ?

File: src/theme/DocPage/index.tsx [hunk 3]
```diff
@@ -60,68 +58,23 @@ function DocPageContent({
...
-              styles.docItemWrapper,
-              {
-                [styles.docItemWrapperEnhanced]: hiddenSidebarContainer,
-              },
-            )}
-          >
-            <MDXProvider components={MDXComponents}>{children}</MDXProvider>
-          </div>
-        </main>
+      <div className={clsx('grid grid-cols-12 gap-x-5 mx-auto')} style={{ maxWidth: '1120px' }}>
```
Comment (RomanAchilov) [inline src/theme/DocPage/index.tsx]: please leave a comment why maxWidth is needed

**Review by RomanAchilov — CHANGES_REQUESTED** [review]

File: src/theme/DocPage/index.tsx (hunk 3 above)
Comment (IliaSolovev) [reply src/theme/DocPage/index.tsx]: I fix the width of the container like on mockup

File: src/theme/DocItem/styles.scss (hunk 2 above)
Comment (IliaSolovev) [reply src/theme/DocItem/styles.scss:13]: it on mockup

File: src/theme/DocItem/index.tsx (hunk 1 above)
Comment (RomanDavlyatshin) [reply src/theme/DocItem/index.tsx]: As far as I understand, it is pointless to use (neither harmful nor useful) `<header>` element to wrap a single `<h*>` tag.
According to [the spec](https://html.spec.whatwg.org/multipage/sections.html#the-header-element) "_The header element represents a group of introductory or navigational aids_".
In our case, there is no group, thus, `<header>` element can be omitted.
I suggest removing it.

File: src/theme/DocSidebar/styles.scss [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+      }
+
+      &:hover {
+        background: map-get($monochrome, light-tint);
+      }
+    }
+
+    .menu__list {
+      margin-left: 0;
+      overflow-y: hidden;
```
Comment (RomanDavlyatshin) [inline src/theme/DocSidebar/styles.scss]: What's the reason behind adding that?

File: src/theme/DocPage/index.tsx (hunk 3 above)
Comment (RomanDavlyatshin) [reply src/theme/DocPage/index.tsx]: Docs page layout is broken on `480` screen:
1. Menu does not hide under burger menu anymore (compare to docusaurus)
2. TOC just disappears (again, compare to docusaurus page)
image

image

Comment (RomanDavlyatshin) [reply src/theme/DocPage/index.tsx]: A also believe we should set `max-width` for containers via media queries, and not just use a static value.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]
https://github.com/Drill4J/drill4j-website/pull/29/files#r618166808

File: src/theme/DocSidebar/styles.scss (hunk 4 above)
Comment (IliaSolovev) [reply src/theme/DocSidebar/styles.scss]: We can see scroll because it has transition

**Review by RomanAchilov — APPROVED** [review]

File: src/hooks/use-breakpoint.ts [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+  lg: 1024,
+  xl: 1280,
+  '2xl': 1536,
+};
+
+export const useBreakpoint = (size: keyof typeof windowSizes) => {
+  const [isWindowLessSize, setIsWindowLessSize] = useState(false);
+  useEffect(() => {
+    function handleResize() {
+      if (window.innerWidth < windowSizes[size]) {
```
Comment (RomanDavlyatshin) [inline src/hooks/use-breakpoint.ts]: I would suggest implementing a check using `>=` instead, to match `@media (min-width: *)` behaviour used in css classes

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

File: src/theme/DocItem/index.tsx [hunk 6]
```diff
@@ -36,7 +36,7 @@ function DocItem(props: Props): JSX.Element {
     title,
     permalink,
   } = metadata;
-  const isWindowLessLarge = useBreakpoint('lg');
+  const isWindowLarge = useBreakpoint('lg');
```
Comment (RomanDavlyatshin) [inline src/theme/DocItem/index.tsx]: These names could get a bit long and confusing in terms of "what is what" (lg==Large, Xl==?).
I suggest changing to the `isBreakpoint__` format.
E.g.: `isBreakpointLg`, `isBreakpointXl`, `isBreakpointSm`?

File: src/hooks/use-click-outside.ts [hunk 7]
```diff
@@ -0,0 +1,27 @@
+import { useRef, useEffect } from 'react';
+
+export function useClickOutside(callback: () => void) {
```
Comment (RomanDavlyatshin) [inline src/hooks/use-click-outside.ts]: Finally, I can suggest a better name 😃 
How about `useBlur`? It encompasses both clicks and `Esc` keypresses

File: src/theme/Navbar/index.tsx [hunk 8]
```diff
@@ -5,41 +5,71 @@
...
-                  className={items.length !== index + 1
-                    ? 'hidden md:block text-16 font-normal leading-24 text-monochrome-default hover:text-blue-default'
-                    : 'button-primary md:ml-5'}
-                  to={useBaseUrl(to)}
-                >
-                  {label}
-                </Link>
-              </li>
-            ))}
+        <section className="md:flex items-center" ref={ref}>
```
Comment (RomanDavlyatshin) [inline src/theme/Navbar/index.tsx]: I don't think that using a `<section>` element here makes sense:

- According to [the spec](https://html.spec.whatwg.org/multipage/sections.html#the-section-element) `... the section element is appropriate only if the element's contents would be listed explicitly in the document's outline.`
- The header menu definitely will not end up in the document's outline.

I suggest changing it to `<div>` instead.

**Review by RomanDavlyatshin — CHANGES_REQUESTED** [review]

**Review by RomanDavlyatshin — APPROVED** [review]

**Pull request merged**